src/pykeen/models/base.py

```diff
@@ -120,69 +347,20 @@ def __init__(
         '''
         self.predict_with_sigmoid = predict_with_sigmoid
 
-    def __init_subclass__(cls, autoreset: bool = True, **kwargs):  # noqa:D105
-        cls._is_base_model = not autoreset
-        if not cls._is_base_model:
-            _add_post_reset_parameters(cls)
-
-    @property
-    def can_slice_h(self) -> bool:
-        """Whether score_h supports slicing."""
-        return _can_slice(self.score_h)
-
-    @property
-    def can_slice_r(self) -> bool:
-        """Whether score_r supports slicing."""
-        return _can_slice(self.score_r)
-
-    @property
-    def can_slice_t(self) -> bool:
-        """Whether score_t supports slicing."""
-        return _can_slice(self.score_t)
-
-    @abstractmethod
-    def _reset_parameters_(self):  # noqa: D401
-        """Reset all parameters of the model in-place."""
-        raise NotImplementedError
-
-    def reset_parameters_(self) -> 'Model':  # noqa: D401
+    def reset_parameters_(self):  # noqa: D401
         """Reset all parameters of the model and enforce model constraints."""
         self._reset_parameters_()
         self.to_device_()
         self.post_parameter_update()
         return self
 
-    @property
-    def num_entities(self) -> int:  # noqa: D401
-        """The number of entities in the knowledge graph."""
-        return self.triples_factory.num_entities
-
-    @property
-    def num_relations(self) -> int:  # noqa: D401
-        """The number of unique relation types in the knowledge graph."""
-        return self.triples_factory.num_relations
-
-    def _set_device(self, device: DeviceHint = None) -> None:
-        """Set the Torch device to use."""
-        self.device = resolve_device(device=device)
-
-    def to_device_(self) -> 'Model':
+    def to_device_(self):
         """Transfer model to device."""
         self.to(self.device)
         self.regularizer.to(self.device)
```

Comment:
If regularizer subclasses from nn.Module, pytorch will take care of moving it along the model.

Reply:
Excellent. Addressed in 363b0eb7